socorro/analysis/correlations/correlations_app.py

```diff
@@ -73,21 +67,82 @@ class CorrelationsApp(FetchTransformSaveWithSeparateNewCrashSourceApp):
         likely_to_be_changed=True,
     )
 
+    required_config.add_option(
+        'transaction_executor_class',
+        default="socorro.database.transaction_executor."
+        "TransactionExecutorWithInfiniteBackoff",
+        doc='a class that will manage transactions',
+        from_string_converter=class_converter,
+        reference_value_from='resource.postgresql',
+    )
+    required_config.add_option(
+        'database_class',
+        default=(
+            'socorro.external.postgresql.connection_context'
+            '.ConnectionContext'
+        ),
+        doc='the class responsible for connecting to Postgres',
+        from_string_converter=class_converter,
+        reference_value_from='resource.postgresql',
+    )
+
+    required_config.add_option(
+        name='date',
+        doc='Specific date to run this for',
+        default='',
+        from_string_converter=date_with_default_yesterday,
+    )
+
+    required_config.add_option(
+        name='product',
+        doc='Product name',
+        default='Firefox',
+    )
+
+    #--------------------------------------------------------------------------
+    def __init__(self, config, quit_check_callback=None):
+        super(CorrelationsApp, self).__init__(config)
+
     #--------------------------------------------------------------------------
     @staticmethod
     def get_application_defaults():
         return {
             "number_of_submissions": 'all',
-            "source.crashstorage_class":
-                'socorro.external.boto.crashstorage.BotoS3CrashStorage',
-            "destination.crashstorage_class":
-                'socorro.external.crashstorage_base.NullCrashStorage',
-            "new_crash_source.new_crash_source_class":
-                'socorro.external.postgresql.new_crash_source'
-                '.PGPVNewCrashSource',
+            "source.crashstorage_class": (
+                'socorro.external.boto.crashstorage.BotoS3CrashStorage'
+            ),
+            "destination.crashstorage_class": (
+                'socorro.external.crashstorage_base.NullCrashStorage'
+            ),
+            "new_crash_source.new_crash_source_class": (
+                'socorro.external.es.new_crash_source.ESNewCrashSource'
+            ),
         }
 
     #--------------------------------------------------------------------------
+    def _create_iter(self):
+        hits = ProductVersions(config=self.config).get(
+            active=True,
+            product=self.config.product
+        )['hits']
+        versions = [
+            x['version'] for x in hits if not x['version'].endswith('b')
+        ]
+        assert versions, "No active versions"
+
+        # convert a datetime.date object to datetime.datetime
+        dt = datetime.datetime(
+            self.config.date.year,
+            self.config.date.month,
+            self.config.date.day,
+        ).replace(tzinfo=UTC)
```

Comment:
I kinda like that the config is all about a date. But the internal implementation is a datetime. 
If you do `socorro correlations --date=2016-03-28 --help` it will display what value we have for the date and if we let the config be a datetime it'd now display a datetime which isn't what I put in.